Structure_Engine/Modify/SetElements0D.cs

```diff
@@ -49,9 +50,27 @@ public static Bar SetElements0D(this Bar bar, List<IElement0D> newElements0D)
                 return null;
             }
 
-            Bar clone = bar.GetShallowClone() as Bar;
-            clone.StartNode = newElements0D[0] as Node;
-            clone.EndNode = newElements0D[1] as Node;
+            Bar clone = bar.DeepClone() as Bar;
+
+            // Default the Bars end if the input is an Point
+            if (newElements0D[0] is Point)
+            {
+                clone.StartNode = Create.Node(newElements0D[0] as Point);
+                if (clone.Release != null)
+                    clone.Release.StartRelease = Create.FixConstraint6DOF();
+            } else
```

Comment:
final thing, being really picky, can we add a new line between the "}" and the "else"